_posts/2019-02-15-snapshots-of-buggy-black-boxes.md

```diff
@@ -0,0 +1,128 @@
+---
+layout: post
+title: Snapshots of Buggy Black Boxes
+author: mpajunen
+excerpt: Snapshot testing has been gaining popularity in frontend development. Could it be useful for testing legacy code as well?
+tags:
+- legacy code
+- testing
+---
+
+Snapshot tests have recently become popular in testing UI components. We decided to try snapshots in testing legacy code.
+
+## A Black Box of Code
+
+A while ago we were faced with a typical situation when working with legacy code: A largely unknown and practically untested Black Box. This Black Box had been lifted from an older system and rewritten for the new platform already. Unfortunately, much of the old cruft remained.
+
+We already had a list of new features and improvements to implement. Looking at the code this seemed like a tall order: Data fetches, business logic and output marshalling were all gloriously mixed up. It didn't take long to realize that significant refactoring was needed before getting to the new features.
```

Comment:
refactoring was needed before we could start with the new features?

Reply:
refactoring was needed before we could start working on the new features